src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Usage/DetectPLINQNops.Fixer.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.NetCore.Analyzers;
+using Microsoft.NetCore.Analyzers.Usage;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Usage
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = DetectPLINQNopsAnalyzer.RuleId), Shared]
+    public sealed class DetectPLINQNopsFixer : CodeFixProvider
+    {
+        private static readonly string[] removableEnds = new string[] { "ToList", "ToArray", "AsParallel", "ToDictionary", "ToHashSet" };
+
+        private static readonly string[] requiredAppendableEnds = new string[] { "ToDictionary", "ToHashSet" };
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(DetectPLINQNopsAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            var node = root.FindNode(context.Span);
+            if (node is not InvocationExpressionSyntax declaration)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(
+                new AsParallelCodeAction(
+                    title: MicrosoftNetCoreAnalyzersResources.RemoveRedundantCall,
```

Comment:
Not sure we do want to reuse this message or create a new specific one.

Reply:
I have no idea, I'll adapt this to whatever is decided. It seemed to accurately reflect what the code fix does so I just chose that